src/front/glsl/ast.rs

```diff
@@ -203,12 +203,15 @@ impl<'function> Context<'function> {
 
         for &(ref name, lookup) in program.global_variables.iter() {
             this.emit_flush(body);
-            let expr = match lookup {
-                GlobalLookup::Variable(v) => Expression::GlobalVariable(v),
+            let (expr, load) = match lookup {
+                GlobalLookup::Variable(v) => (
+                    Expression::GlobalVariable(v),
+                    StorageClass::Handle != program.module.global_variables[v].class,
```

Comment:
nit: doing the "constant != variable" comparison is super weird. Can we please use the normal order of things?

Reply:
And what would be the normal order? An equality comparison?

Reply:
"variable != something" or "variable == something"
I think the "constant == variable" comes from some very old times where people were afraid to accidentally write assignment expression instead of equality (in C/C++).

Reply:
Sorry I'm really not understanding what you want here, do you want me to change the order of the comparsion i.e.
```rust
program.module.global_variables[v].class != StorageClass::Handle
```

Reply:
Glad we managed to get on the same page